crates/ark/src/lsp/backend.rs

```diff
@@ -381,8 +451,10 @@ pub fn start_lsp(runtime: Arc<Runtime>, address: String, conn_init_tx: Sender<bo
         log::trace!("Connected to LSP at '{}'", address);
         let (read, write) = tokio::io::split(stream);
 
+        let (shutdown_tx, mut shutdown_rx) = tokio::sync::mpsc::channel::<()>(1);
+
         let init = |client: Client| {
-            let state = GlobalState::new(client);
+            let state = GlobalState::new(client.clone());
```

Comment:
```suggestion
            let state = GlobalState::new(client);
```

When you remove `client` from `Backend` can also go back to this